tiledb/cf/core.py

```diff
@@ -384,107 +385,54 @@ def __exit__(self, exception_type, exception_value, exception_traceback):
-
-    def array_schema(self, array_name):
-        return self.group_schema.get(array_name)
-
-    def array_metadata(self, array_name: str) -> ArrayMetadata:
-        with self.get_open_uri_key(array_name) as array:
-            meta = ArrayMetadata(array.meta)
-        return meta
-
-    def attr_metadata(self, attr: str) -> Union[AttrMetadata, Dict[str, AttrMetadata]]:
-        """Attribute metadata wrapper for the attribute accessed at initialization.
-
-        Can only be used if an attribute or an array with exactly one attribute was
-        specified when initializing this group.
-        """
-        attr_metadata = dict()
-        array_names = self.group_schema.arrays_with_attr(attr)
-        if len(array_names) == 0:
-            return "no attr metadata found"
-        for name in array_names:
-            if name != METADATA_ARRAY_NAME:
-                with self.get_open_uri_key(name) as array:
-                    metadata = AttrMetadata(array.meta, attr)
-                    attr_metadata[name] = metadata
-        if len(array_names) == 1:
-            return attr_metadata[array_names[0]]
-        return attr_metadata
+        if self._metadata_array is not None:
+            self._metadata_array.close()
+        for array in self._opened_arrays:
+            array.close()
+        self._opened_arrays.clear()
 
     @property
     def has_metadata_array(self) -> bool:
         """Flag that is true if there a metadata array for storing group metadata."""
-        return self.metadata_array is not None
+        return self._metadata_array is not None
 
     @property
     def meta(self) -> Optional[tiledb.Metadata]:
         """Metadata object for the group, or ``None`` if no array to store group
         metadata exists."""
-        if self.metadata_array is None:
+        if self._metadata_array is None:
             return None
-        with self.metadata_array as meta_array:
-            meta = Metadata(meta_array.meta)
-        return meta
+        return self._metadata_array.meta
+
+    def open_array(
+        self, array: Optional[str] = None, attr: Optional[str] = None
+    ) -> tiledb.Array:
+        if array is None and attr is None:
+            raise ValueError(
+                "Cannot open array. Either an array or attribute must be specified."
+            )
+        if array is None:
+            array_names = self._group_schema.arrays_with_attr(attr)
+            if array_names is None:
+                raise KeyError(f"No attribute with name '{attr}' found.")
+            if len(array_names) > 1:
+                raise ValueError(
+                    f"The array must be specified when opening an attribute that "
+                    f"exists in multiple arrays in a group. Arrays with attribute "
+                    f"'{attr}' include: {array_names}."
+                )
+            array = array_names[0]
+        tiledb_array = tiledb.open(
+            self._array_uris[array],
+            mode=self._mode,
```

Comment:
Could we specify mode more flexibly in the open_array method rather than once globally for the Group

Reply:
I personally prefer having the mode set to be consistent with the group, and it breaks the current API less to leave it that way. How about adding an optional mode that if None, inherits the group mode?

Reply:
That sounds cool